packages/ember-engines/addon/-private/engine-instance-ext.js

```diff
@@ -48,6 +63,11 @@ EngineInstance.reopen({
     this._externalRoutes = {};
   },
 
+  lookup(fullName, options) {
+    deprecateHostRouterService(fullName);
```

Comment:
This now triggers the deprecation warning for _every_ lookup.

Also, in case of doing `lookup('service:router')`, the deprecation message makes it seem like you have to do `lookup('hostRouter')` or `lookup('appRouter')` instead.

Maybe it's better to have separate deprecations/messages for these?

Additionally, in https://github.com/ember-engines/ember-engines/pull/759, a setup was added for testing deprecations.

Reply:
Thanks a lot! I'll a condition for that, I lost it on refactoring

> Maybe it's better to have separate deprecations/messages for these?

Why we need separate messages? The users are trying to get the app router from the root application directly. The message should be the same, warn them to avoid inject `router` directly without any alias.

Reply:
This:
```js
fullName === 'router'
```
Should be:
```js
fullName === 'service:router'
```
`router` is not a full name.

I think the current tests are passing by accident actually. The last two assertions pass because the deprecation was already captured before. If you change `blogEngineInstance.lookup('service:router');` to `blogEngineInstance.lookup('service:blablabla');` the tests still pass. So I think the current test is not reliable.

> Why we need separate messages? The users are trying to get the app router from the root application directly.

Yes, the user requests access to the host's router in both cases, but _the way_ the user does it in both scenario's (defining a dependency vs doing a manual lookup) is different and I feel that using the same message to cover both scenario's isn't clear.

That could just be me of course. I still feel like the current deprecation message doesn't really fit the lookup scenario: `Please use a different name as 'hostRouter' or 'appRouter' instead of 'service:router'.` makes it seem that the user needs to do `lookup('hostRouter')` instead which isn't the case.

Reply:
> he tests still pass. So I think the current test is not reliable.

I agree, what's happening here is a cache in `assert.deprecationsInclude` the extension on Qunit that you added on #759 has a bug. 

> Yes, the user requests access to the host's router in both cases, but the way the user does it in both scenario's (defining a dependency vs doing a manual lookup) is different and I feel that using the same message to cover both scenario's isn't clear.

I don't agree here! We shouldn't say anything about `lookup` because the first thing to do is declare `router` as an alias like `{ 'app-router': 'router' }`, then we should not suggest to the users use `lookup('hostRouter')` directly jumping the first step (`{ 'app-router': 'router' }`). The Ember.js users already know everything about lookup and how to use that with different names. Then, IMO the deprecate message should be the same, given we're deprecating the same "feature" in different levels.

Reply:
It's not a bug 😀. `assert.deprecationsInclude` should **not** change the state during a single test.
It should only check if a deprecation is included or not. Instead, you could extend the method
to check if a deprecation is included X times.

For example (just a suggestion):
```js
assert.deprecationsInclude(deprecationMessage, { count: 2 });
```
Simply removing the other assertions does not make the test reliable again.

**Regarding the lookup:**

Sorry, I have a hard time understanding what you are trying to say. I'll try to explain myself better.

It _is_ possible that the user does a lookup of `service:router` **without** defining the dependency.
As mentioned in [this comment](https://github.com/ember-engines/ember-engines/pull/764#issuecomment-816721788) the router service is injected automatically.
In this scenario I still feel that the current deprecation message is not clear enough.
The guide only covers the case where the user has already defined the dependency.

I am **not** saying we should suggest using `lookup('hostRouter')`.
I am only saying that the current deprecation **makes it seem** like the user should.
Which I feel can be confusing.

Additionally, the deprecation is now also triggered for `router:main`.
**even though the user isn't doing this lookup themselves**.
So there's no way for the user to get rid of the deprecation in this case.
Again, this is because the router service is already injected automatically (I think).
Manually defining the dependency **with or without** an alias won't resolve the deprecation here.
You can test this in the dummy app. The `ember-blog` does not have a dependency on the router service and it also does not do a lookup of the router service, yet the deprecation message is displayed.

Reply:
> Simply removing the other assertions does not make the test reliable again.

I just removed them because they're not possible to test. And if you put `{ count: 25 }` or any other number it will continue passing. The problem here is because this assertion is not reliable. 


```js
return blogEngineInstance.boot().then(() => {
    blogEngineInstance.lookup('service:router');

    assert.deprecationsInclude(
      `Support for the host's router service has been deprecated. Please use a different name as 'hostRouter' or 'appRouter' instead of 'router'.`,
    { count: 2 });

    // app router from root application
    blogEngineInstance.lookup('router:main');

    assert.deprecationsInclude(
      `Support for the host's router service has been deprecated. Please use a different name as 'hostRouter' or 'appRouter' instead of 'router'.`,
    { count: 7 });
  });
 ```
  
  > Additionally, the deprecation is now also triggered for router:main.
even though the user isn't doing this lookup themselves.

Many users use `router:main` to get the router from host application. Then it's not true!

Reply:
I have tested this myself, otherwise I would not suggest it as a possible solution. I can understand that it's a bit harder to test, but it's not impossible. You would have to update the test setup yourself for it to work though.

> Many users use `router:main` to get the router from the host application.

I did not say that there aren't any users that do this.
I said that, at the moment the deprecation is also triggered without looking up `router:main` manually, you can see this when running the dummy app.
Triggering the deprecation for logic over which the user has no control seems strange.
There's also already a [lint rule](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-private-routing-service.md) for this case, so maybe triggering the deprecation for `router:main` is not needed?

Regarding the lookup, I tested this again and indeed, it seems the lookup returns `undefined` for the router service.
I must have been mistaken the first time I tested it, so you are right.

I'm feeling some frustration on your side regarding my feedback, so I'm backing off as I feel this is adding too much noise to the PR. Thanks for working on this though 👍.

Reply:
> I'm feeling some frustration on your side regarding my feedback

I'm not frustrated. It's a pleasure for me to teach about Ember Engines. Thank you so much for challenging me to write the tests here and improve the code 😄